framework/src/base/MeshGeneratorSystem.C

```diff
@@ -645,3 +644,38 @@ MeshGeneratorSystem::hasDataDrivenAllowed() const
 {
   return _app.parameters().get<bool>(allow_data_driven_param);
 }
+
+bool
+MeshGeneratorSystem::checkUpstreamGenerator(const std::string current_generator,
+                                            const std::string generator_to_compare) const
+{
+  bool is_upstream = false;
+  bool found_generator_to_compare = false;
+  bool found_current_generator = false;
+  for (const auto & generator_names : _ordered_generators)
+  {
+    // First check if current_generator is defined at current dependency level
+    if (std::find(generator_names.begin(), generator_names.end(), current_generator) !=
+        generator_names.end())
```

Comment:
you ll just need a custom comparator here to be able to use the other vector (with pointers to MGs instead of names)

Reply:
Related to my other comment, but I don't think pointers to the MG are defined at the point where the constructor is called?